persistence/rest/src/main/java/org/infinispan/persistence/rest/RestStore.java

```diff
@@ -257,51 +313,59 @@ else if (timeout > 0 && timeout < 1000)
          return TimeUnit.MILLISECONDS.toSeconds(timeout);
    }
 
-   private long timeHeaderToSeconds(Header header) {
-      return header == null ? -1 : Long.parseLong(header.getValue());
+   private long timeHeaderToSeconds(String header) {
+      return header == null ? -1 : Long.parseLong(header);
    }
 
 
    @Override
    public void process(KeyFilter keyFilter, final CacheLoaderTask cacheLoaderTask, Executor executor, boolean loadValue, boolean loadMetadata) {
-      HttpGet get = new HttpGet(path + "?global");
-      get.addHeader(HttpHeaders.ACCEPT, "text/plain");
-      get.addHeader(HttpHeaders.ACCEPT_CHARSET, "UTF-8");
+      DefaultHttpRequest get = new DefaultHttpRequest(HttpVersion.HTTP_1_1, HttpMethod.GET, path + "?global");
+      get.headers().add(HttpHeaders.Names.ACCEPT, "text/plain");
+      get.headers().add(HttpHeaders.Names.ACCEPT_CHARSET, "UTF-8");
       try {
-         HttpResponse response = httpClient.execute(httpHost, get);
-         HttpEntity entity = response.getEntity();
+         HttpResponseHandler handler = new HttpResponseHandler(true);
+         Channel ch = bootstrap.connect(configuration.host(), configuration.port()).awaitUninterruptibly().channel().pipeline().addLast(
+               new HttpObjectAggregator(maxContentLength), handler).channel();
+         ch.writeAndFlush(get).sync().channel().closeFuture().sync();
          int batchSize = 1000;
          ExecutorAllCompletionService eacs = new ExecutorAllCompletionService(executor);
          final TaskContext taskContext = new TaskContextImpl();
-         BufferedReader reader = new BufferedReader(new InputStreamReader(entity.getContent(), "UTF-8"));
-         Set<Object> entries = new HashSet<Object>(batchSize);
-         for (String stringKey = reader.readLine(); stringKey != null; stringKey = reader.readLine()) {
-            Object key = key2StringMapper.getKeyMapping(stringKey);
-            if (keyFilter == null || keyFilter.accept(key))
-               entries.add(key);
-            if (entries.size() == batchSize) {
-               final Set<Object> batch = entries;
-               entries = new HashSet<Object>(batchSize);
-               submitProcessTask(cacheLoaderTask, eacs, taskContext, batch, loadValue, loadMetadata);
+         try {
+            BufferedReader reader = new BufferedReader(new InputStreamReader(new ByteBufInputStream(((FullHttpResponse) handler.getResponse()).content())));
```

Comment:
The code you replaced specified a specific encoding. Is this not required here ?